Fix crash on addIceCandidate, setRemote/LocalDescription

mininny opened the pull request.

intoxicated commented:
* -> std::unique_ptr 로 바꾼 이유가 있나요? objectivec++ 에서는 * 하면 아직도 alloc / release 해줘야하나요?

mininny commented:
> * -> std::unique_ptr 로 바꾼 이유가 있나요? objectivec++ 에서는 * 하면 아직도 alloc / release 해줘야하나요?

> Second, unnecessary sLD/sRD calls Clone() of the raw pointer and does not delete it, so this pointer will leak. To solve these problems, I changed the return type of nativeDescription to std::unique_ptr and removed the call to Clone() method.

저 타입이 c++타입이라서 그렇습니당

jaesung-0o0 commented on sdk/objc/unittests/RTCPeerConnectionTest.mm:
```diff
@@ -137,6 +141,66 @@ - (void)testWithDependencies {
+                     completionHandler:^(NSError *error) {
+                       ASSERT_NE(error, nil);
+                       if (error != nil) {
+                         dispatch_semaphore_signal(negotiatedSem);
+                       }
+                     }];
+
+  NSTimeInterval timeout = 5;
+  ASSERT_EQ(
+      0,
+      dispatch_semaphore_wait(negotiatedSem,
+                              dispatch_time(DISPATCH_TIME_NOW, (int64_t)(timeout * NSEC_PER_SEC))));
```
167 번째줄의 ASSERT_EQ은 162번째줄의 disaptch_semaphor_signal 메소드 호출까지 기다리는게 맞나요?

jaesung-0o0 commented on sdk/objc/unittests/RTCSessionDescriptionTest.mm:
```diff
@@ -31,7 +31,7 @@ - (void)testSessionDescriptionConversion {
       [[RTC_OBJC_TYPE(RTCSessionDescription) alloc] initWithType:RTCSdpTypeAnswer sdp:[self sdp]];
 
-  webrtc::SessionDescriptionInterface *nativeDescription =
+  std::unique_ptr<webrtc::SessionDescriptionInterface> nativeDescription =
```
둘의 차이가 뭔가요?

jaesung-0o0 commented on sdk/objc/api/peerconnection/RTCPeerConnection.mm:
```diff
@@ -581,7 +579,7 @@ - (void)setLocalDescription:(RTC_OBJC_TYPE(RTCSessionDescription) *)sdp
   rtc::scoped_refptr<webrtc::SetLocalDescriptionObserverInterface> observer(
       new rtc::RefCountedObject<::SetSessionDescriptionObserver>(completionHandler));
-  _peerConnection->SetLocalDescription(sdp.nativeDescription->Clone(), observer);
+  _peerConnection->SetLocalDescription(sdp.nativeDescription, observer);
```
Clone()은 뭔가요? Copying인가요?
제거하신 이유가 뭔가요?
